src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/custom.py

```diff
@@ -263,8 +271,28 @@ def update_extension(extension_name, index_url=None, pip_extra_index_urls=None,
         raise CLIError(e)
 
 
-def list_available_extensions(index_url=None):
-    return get_index_extensions(index_url=index_url)
+def list_available_extensions(index_url=None, show_details=False):
+    index_data = get_index_extensions(index_url=index_url)
+    if show_details:
+        return index_data
+    installed_extensions = get_extensions()
+    installed_extension_names = [e.name for e in installed_extensions]
+    results = []
+    for name, items in OrderedDict(sorted(index_data.items())).items():
+        latest = sorted(items, key=lambda c: parse_version(c['metadata']['version']), reverse=True)[0]
+        installed = False
+        if name in installed_extension_names:
+            installed = True
+            if parse_version(latest['metadata']['version']) > parse_version(get_extension(name).version):
+                installed = str(True) + ' (upgrade available)'
+        results.append({
+            'name': name,
+            'version': latest['metadata']['version'],
+            'summary': latest['metadata']['summary'],
+            'preview': latest['metadata'].get(EXT_METADATA_ISPREVIEW, False),
+            'installed': installed
+        })
+    return results
```

Comment:
Do we have a test for this?

Reply:
I do have some tests but not for these changes...
I will take a look at adding tests... 👍